xapian-core/include/xapian/cluster.h

```diff
@@ -28,14 +28,45 @@
 #endif
 
 #include <xapian/mset.h>
+#include <xapian/queryparser.h>
 #include <xapian/types.h>
 #include <xapian/visibility.h>
 
 #include <unordered_map>
+#include <unordered_set>
 #include <vector>
 
 namespace Xapian {
 
+/// Stopper subclass which checks for both stemmed and unstemmed stopwords
+class XAPIAN_VISIBILITY_DEFAULT StemStopper : public Xapian::Stopper {
+  public:
+    /// Stemming strategies
+    typedef enum { STEM_NONE, STEM_SOME, STEM_ALL_Z } stem_strategy;
+
+    /** Constructor
+     *
+     *  @param stemmer	The Xapian::Stem object to set.
+     */
+    explicit StemStopper(const Xapian::Stem &stemmer);
+
+    std::string get_description() const;
+
+    bool operator()(const std::string & term) const {
+	return stop_words.find(term) != stop_words.end();
+    }
+
+    /// Add a single stop word and its stemmed equivalent
+    void add(const std::string &term);
+
+    void set_stemming_strategy(stem_strategy strategy);
```

Comment:
It probably makes more sense to make this an optional parameter on the constructor (default `STEM_SOME`).  It doesn't seem useful to be able to change it (like it's not useful to change the stemmer used for a given `StemStopper` object), and more convenient to be able to set up the stopper in one go.